rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,103 @@
+
+```yaml
+kind: node
+sub_kind: openssh
+metadata:
+  name: 5da56852-2adb-4540-a37c-80790203f6a9
+spec:
+  addr: 1.2.3.4:22
+  hostname: agentless-node
+version: v2
+```
+
+### OpenSSH CA
+
+For security related reasons that will be discussed below, a new CA will be added called OpenSSH CA. The OpenSSH CA will be responsible for generating and signing user certificates that will be used to authenticate with registered OpenSSH nodes. The public key of the OpenSSH CA will be copied to all registered OpenSSH nodes and configured as `TrustedUserCAKeys` in `sshd_config`.
+
+#### Manual certificate rotation
+
+Users not using EC2 discovery mode and hosting registered OpenSSH nodes themselves will have to rotate the OpenSSH CA manually. They can do so by:
+
+1. Exporting the OpenSSH CA: `tctl auth export --type=openssh | sed "s/cert-authority\ //"`
+2. Writing the exported CA to the configuration folder of the node's SSH daemon, ex. `/etc/ssh/` for OpenSSH on Linux
+3. Replace the `TrustedUserCAKeys` setting in `sshd_config` to point to the newly created CA
+
+### RBAC
+
+
+When a user sends a request to a Proxy to connect to a node, the Proxy will attempt to find a node resource by either its hostname or IP, whichever the user specified. If the resource exists and has the `openssh` `sub_kind`, an RBAC check will be performed. If the resource does not exist or isn't an `openssh` node, the connection flow will continue as normal.
+
+### Security
+
+Both RBAC checks and session recording for registered OpenSSH nodes require that users connect through a Proxy. If users are able to connect to registered OpenSSH nodes directly, they can bypass both features. Currently the User CA public key is copied to OpenSSH nodes and configured to be trusted by `sshd`. This means OpenSSH nodes will accept certificates that are signed by the User CA. The problem with that is when Teleport users authenticate with a cluster, the Auth server replies with a certificate that is signed by the User CA. Users could potentially use this certificate to directly connect to registered OpenSSH nodes.
+
+Using the new OpenSSH CA to sign certificates used to authenticate with registered OpenSSH nodes solves this problem, as Teleport users do not have access to any certificates signed by the OpenSSH CA.
+
+### UX
+
+The following Teleport node features won't work with registered OpenSSH nodes:
+
+- Enhanced session recording and restricted networking
+- Host user provisioning
+- Session recording without SSH session termination
+- Dynamic labels
+- Outbound persistent tunnels to Proxies
+
+Due to this and other potential future differences, `tsh ls` and the node listing on the web UI should be updated to display if nodes are registered OpenSSH nodes or Teleport agent nodes.
+
+### Future work
```

Comment:
I don't think that anything mentioned below is "Future work".

Reply:
Done